usaspending_api/etl/es_etl_helpers.py

```diff
@@ -290,78 +306,91 @@ def es_data_loader(client, fetch_jobs, done_jobs, config):
         raise SystemExit(1)
 
     printf({"msg": "Success: {}, Fails: {}".format(success, failed), "job": job_id, "f": "ES Ingest"})
     return success, failed
 
 
-def put_alias(client, index, alias_name, award_type_codes):
-    alias_body = {"filter": {"terms": {"type": award_type_codes}}}
+def put_alias(client, index, alias_name, alias_body):
     client.indices.put_alias(index, alias_name, body=alias_body)
 
 
 def create_aliases(client, index, silent=False):
     for award_type, award_type_codes in INDEX_ALIASES_TO_AWARD_TYPES.items():
-        alias_name = "{}-{}".format(settings.TRANSACTIONS_INDEX_ROOT, award_type)
+        alias_name = "{}-{}".format(settings.ES_TRANSACTIONS_READ_ALIAS_PREFIX, award_type)
         if silent is False:
             printf(
                 {
-                    "msg": 'Putting alias "{}" with award codes {}'.format(alias_name, award_type_codes),
-                    "job": "",
+                    "msg": "Putting alias '{}' on {} with award codes {}".format(alias_name, index, award_type_codes),
+                    "job": None,
                     "f": "ES Alias Put",
                 }
             )
-        put_alias(client, index, alias_name, award_type_codes)
+        alias_body = {"filter": {"terms": {"type": award_type_codes}}}
+        put_alias(client, index, alias_name, alias_body)
+
+    # ensure the new index is added to the alias used for incremental loads.
+    # If the alias is on multiple indexes, the loads will fail!
+    printf(
+        {
+            "msg": "Putting alias '{}' on {}".format(settings.ES_TRANSACTIONS_WRITE_ALIAS, index),
+            "job": None,
+            "f": "ES Alias Put",
+        }
+    )
+    put_alias(client, index, settings.ES_TRANSACTIONS_WRITE_ALIAS, {})
 
 
-def swap_aliases(client, index):
+def set_final_index_config(client, index):
+    es_settingsfile = str(settings.APP_DIR / "etl" / "es_settings.json")
+    with open(es_settingsfile) as f:
+        settings_dict = json.load(f)
+    final_index_settings = settings_dict["final_index_settings"]
+
+    current_settings = client.indices.get(index)[index]["settings"]["index"]
+
+    client.indices.put_settings(final_index_settings, index)
     client.indices.refresh(index)
-    # add null values to contracts alias
+    for setting, value in final_index_settings.items():
+        message = 'Changing "{}" from {} to {}'.format(setting, current_settings.get(setting), value)
+        printf({"msg": message, "job": None, "f": "ES Settings Put"})
+
+
+def swap_aliases(client, index):
     if client.indices.get_alias(index, "*"):
         printf({"msg": 'Removing old aliases for index "{}"'.format(index), "job": None, "f": "ES Alias Drop"})
         client.indices.delete_alias(index, "_all")
 
-    alias_patterns = settings.TRANSACTIONS_INDEX_ROOT + "*"
+    alias_patterns = settings.ES_TRANSACTIONS_READ_ALIAS_PREFIX + "*"
+    old_indexes = []
```

Comment:
why is this needed?

Reply:
`old_indexes` is to store any pre-existing indexes (hopefully only one) which contained any read aliases. Currently the script attempts to close the old indexes, unfortunately since that functionality is absent the AWS ES managed service this PR alters the step to delete the old indexes.

Reply:
cool. I just didn't spot where it was used so I wanted to make sure it wasn't dead code

Reply:
Lines 379-380 and lines 389-391 in this file